### ISSUE TITLE
Call Deferred.start() in Deferred.asListenableFuture()

### ISSUE THREAD

**Issue by eliptus-google** [google/guava#6545]
Kotlin's `Deferred` can be crated with `CoroutineStart.LAZY`. I've found unsuspecting conversions to `ListenableFuture` with the `Deferred.asListenableFuture()` extension waiting forever for completion. Calling `Deferred.start()` when converting seems to produce the desired result.

**Comment by cpovirk** [comment]
(a year and a half later, sorry...)

I think that this is probably an issue for either [kotlinx.coroutines](https://github.com/Kotlin/kotlinx.coroutines) or perhaps Android Lint (if the API needs to continue to exist but is dangerous in some cases that could be statically detected).

**Comment by eliptus-google** [comment]
Neither of these suggestions seem feasible:
1. `kotlinx.coroutines` accounts for lazy `Deferred` by providing a `start()` method. It is not responsible for conversion from `Deferred` to third-party extensions.
2. At best static analysis can warn that a `Deferred` *might* be lazy when using `Deferred.asListenableFuture()`. This would be a noisy and rarely useful lint.

**Comment by jbduncan** [comment]
@eliptus-google I'd be surprised if Guava was the owner of the `Deferred.asListenableFuture` extension. Where did you find it?

If it's in another third-party library, it sounds like it's their responsibility to either provide a fix or document your workaround.

**Comment by cpovirk** [comment]
Yes, thanks, it sounds like the missing piece here is where the function in question is defined. It's in kotlin.coroutines:

https://github.com/Kotlin/kotlinx.coroutines/blob/ee92d16c4b48345648dcd8bb15f11ab9c3747f67/integration/kotlinx-coroutines-guava/src/ListenableFuture.kt#L201

**Comment by eliptus-google** [comment]
Ah, my bad. It didn't occur to me that the standard library and not Guava wouldn't own `kotlinx-coroutines-guava`.
